src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -175,6 +175,12 @@ public synchronized InAppNotification getNotification(int id, boolean replace) {
         return notif;
     }
 
+    // if a notification was failed to show, add it back to the unseen list so that we
+    // won't lose it
+    public synchronized void handleNotificationDisplayFailure(InAppNotification notif) {
+        mUnseenNotifications.add(notif);
```

Comment:
I don't think we need to check it because there're only 2 functions capable of adding new notifications, and the other one is `reportResults` which gets called when we receive a `decide response`. Also, I don't think we ever want to compose notifications on the device. So I think it's pretty safe to add notifications back without checking.